fondant/component_spec.py

```diff
@@ -71,11 +71,15 @@ def __repr__(self) -> str:
     def fields(self) -> t.Mapping[str, Field]:
         return types.MappingProxyType(
             {
-                name: Field(name=name, type=field["type"])
+                name: Field(name=name, type=Type[field["type"]])
                 for name, field in self._specification["fields"].items()
             }
         )
 
+    @property
+    def additional_fields(self) -> bool:
```

Comment:
What's the use case for this? 

(same question for AdditionalSubsets - why would a component want to deal with subsets it doesn't need?)

Reply:
Have a look at the test examples, hopefully they provide some clarity on this.

Some examples where this is useful:
- additionalSubsets
  - An embedder component adds an embedding subset based on images. It doesn't care if there are additional subsets (eg. captions), so it defines `additionalSubsets` to be `true` (the default)
  - A LAION retrieval component adds rows to the `images`, `captions`, and `embeddings` subsets. It should not allow any additional subsets, as they will no longer be available for every entry in the index. So it defines `additionalSubsets` to be `false`.
- additionalFields
  - A component calculating the `width` and `height` of an image doesn't care if there's other metadata as well, so it defines 'additionalFields` to be `true` (the default)
  - A component that transforms images might invalidate some of the metadata. So it can define `additionalFields` to be `false` and define all the metadata it generates explicitly.

The combination of `additionalSubsets` and `additionalFields` for both input and output provides enough flexibility to unambiguously define the output subsets and their fields.

Reply:
It's also not very clear to me why we would need `Additionalsubsets`. All subsets are going to have to be defined in the ComponentSpecs at some point. So subsets that are not specified should just not be loaded. What does `Additionalsubsets` further enable?

Reply:
Ok, but with `additionalSubsets` being `True` by default, it means that a lot more data gets loaded and provided to the component, which might slow things down? e.g. let's say you have 3 subsets in the manifest, "images", "captions" and "embeddings", and a component that only needs the "images" will get all embeddings as well returned

Reply:
No, we would only load subsets and fields explicitly defined in the component spec. `additionalSubsets` and `additionalFields` are only needed to unambiguously define the output manifest of the component.

Reply:
Have a look at test examples 1, 2, 3, or 4, 5, 6. Their component specs are the same except for differences in `additionalSubsets` and `additionalFields`. Without this information, how would you define which of the output manifests would be the correct one?

Reply:
Ok I see, so they define how the component expects the input and output manifest to look like. That makes sense. Thanks for clarifying!

Reply:
Thanks for clarifying.  Another question, if `additionalSubsets`is False (e.g. clip retrieval job) are we then required to populate every other subsets that's currently available and that's available for all other `sources`?

Reply:
> Ok I see, so they define how the component expects the input and output manifest to look like. That makes sense. Thanks for clarifying!

In a way. It also defines the behavior of the component of course, as the output manifest depends on this.

> Thanks for clarifying. Another question, if `additionalSubsets`is False (e.g. clip retrieval job) are we then required to populate every other subsets that's currently available for all other `sources`?

Not sure if I understand your question. But if `additionalSubsets` is False, subsets not explicitly defined in the component spec will not be propagated to the output manifest.

Reply:
Had another look at the example and I think it's all clear now. Thanks